src/OrchardCore.Modules/OrchardCore.Media/Startup.cs

```diff
@@ -202,6 +202,12 @@ public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder ro
             var mediaOptions = serviceProvider.GetRequiredService<IOptions<MediaOptions>>().Value;
             var mediaFileStoreCache = serviceProvider.GetService<IMediaFileStoreCache>();
 
+            // Secure media file middleware, but only if the feature is enabled.
+            if(serviceProvider.IsSecureMediaEnabled())
+            {
+                app.UseMiddleware<SecureMediaMiddleware>();
+            }
```

Comment:
If I remember correctly, using `Order` was not sufficient because of other module dependencies.

Reply:
From what I see it should work, please check.

Reply:
You shouldn't need to do that. `Order` can be any `int`, even negative.

Reply:
Originally, that causes the middleware to be added before calls to `UseAuthentication` and `UseAuthorization`. That has changed since this PR has been submitted, because `UseAuthentication` is now ordered differently by https://github.com/OrchardCMS/OrchardCore/pull/13181. 
But anyway, the middleware would still be added before the users module and the `UseAuthorization` within that module. I'm not sure about the implications of that ordering.

Reply:
We did have a [lengthy debate about that](https://github.com/OrchardCMS/OrchardCore/pull/13181#discussion_r1091643831) ([and here](https://github.com/OrchardCMS/OrchardCore/pull/13181/files#r1454071386)), suspecting that it can break people's code :D.

You only need this to be added before the base Media `Startup`, i.e. with an `Order` of `-1`.

Reply:
> We did have a [lengthy debate about that](https://github.com/OrchardCMS/OrchardCore/pull/13181#discussion_r1091643831) ([and here](https://github.com/OrchardCMS/OrchardCore/pull/13181/files#r1454071386)), suspecting that it can break people's code :D.

In this case, it makes using `Order` work 😉

> You only need this to be added before the base Media `Startup`, i.e. with an `Order` of `-1`.

Yes, as I said, this will work now. But it still adds our middleware before the authorization middleware added by the user module. This would skip all the default authorization policies, and I'm not convinced that's a good idea.
That's why I said that we would have to change the order of the main media module instead. 

I think it would be more robust to use this little trick here and not to use arbitrary ordering numbers. Any ordering we introduce here may change any other module behavior too.

Reply:
My problem is that this introduces a circular dependency. How does `-1` make it execute also before the authorization middleware if it has an order of -150?

Reply:
Ah OK, got it. Let me think about an alternate solution. Having some kind of extension point in the Media `Startup` to register other middleware could be one, but that sounds like overcomplicating things.

Could you please bring this up on the weekly meeting where you'll demo the feature, as well as https://github.com/OrchardCMS/OrchardCore/pull/15173#discussion_r1536282023, to gather feedback there too?

Reply:
I added https://github.com/OrchardCMS/OrchardCore/issues/15716 to discuss further improvements. I'll keep this workaround until we have another solution.

Reply:
Let's leave this, but link to https://github.com/OrchardCMS/OrchardCore/issues/15716 in a comment here.